scheduler/feasible.go

```diff
@@ -313,6 +314,22 @@ func (c *CSIVolumeChecker) isFeasible(n *structs.Node) (bool, string) {
 			return false, fmt.Sprintf(FilterConstraintCSIPluginMaxVolumesTemplate, vol.PluginID, n.ID)
 		}
 
+		// CSI spec: "If requisite is specified, the provisioned
+		// volume MUST be accessible from at least one of the
+		// requisite topologies."
+		if len(vol.Topologies) > 0 {
+			var ok bool
+			for _, requiredTopo := range vol.Topologies {
```

Comment:
Just checking my understanding of this feature: at this point the volume is already registered, so we don't need to worry about `preferred` vs. `required`. The plugin that handled the registration resolved the supported topologies and we stored them in the state store, so any match here would match the user's request.

Reply:
Exactly!